intents/oversights/simpson_paradox.py

```diff
@@ -0,0 +1,198 @@
+	required_columns.append(metric)
+	table = aspects.crop_other_columns(table, required_columns)
+
+	""" operational_dimensions will contain list of all dimension 
+									except slice_compare_column"""
+	operational_dimensions = all_dimensions.copy()
+	operational_dimensions.remove(slice_compare_column[0])
+	grouping_dimensions = dimensions.copy()
+
+	# slice_compare_column should be the last value of the list
+	grouping_dimensions.remove(slice_compare_column[0])
+	grouping_dimensions.append(slice_compare_column[0])
+
+	required_table = table.copy()
+	required_columns = grouping_dimensions.copy()
+	required_columns.append(metric)
+	required_table = aspects.crop_other_columns(required_table,
+											 required_columns)
+
+	"""initial_result will store the dominent percentage of 
+							initial groups given by user."""
+	initial_result = check_dominent_percentage(required_table, 
+										grouping_dimensions, 
+										slice_compare_column, 
+										summary_operator)
+	simpson_paradox_columns = []
+	max_difference = 75
+	
+	for column in operational_dimensions:
+		new_grouping_dimensions = dimensions.copy()
+		new_grouping_dimensions.remove(slice_compare_column[0])
+		new_grouping_result = 0
+
+		""" if column is already in grouping_dimensions then 
+					we will remove it otherwise we will 
+					add the column to grouping_dimensions"""
+		if (column in grouping_dimensions):
+			new_grouping_dimensions.remove(column)
+			new_grouping_dimensions.append(slice_compare_column[0])
+
+			required_table = table.copy()
+			required_columns = new_grouping_dimensions.copy()
+			required_columns.append(metric)
+			required_table = aspects.crop_other_columns(required_table, 
+														required_columns)
+
+			new_grouping_result = check_dominent_percentage(required_table, 
+											new_grouping_dimensions.copy(), 
+											slice_compare_column, 
+											summary_operator)
+		else:
+			new_grouping_dimensions.append(column)
+			new_grouping_dimensions.append(slice_compare_column[0])
+
+			required_table = table.copy()
+			required_columns = new_grouping_dimensions.copy()
+			required_columns.append(metric)
+			required_table = aspects.crop_other_columns(required_table, 
+														required_columns)
+
+			new_grouping_result = check_dominent_percentage(required_table, 
+													new_grouping_dimensions, 
+													slice_compare_column, 
+													summary_operator)
+
+		if abs(new_grouping_result - initial_result) >= max_difference:
+			max_difference = abs(new_grouping_result - initial_result)
+			simpson_paradox_columns = new_grouping_dimensions
+	if len(simpson_paradox_columns) > 0:
+		suggestion = str(simpson_paradox_columns)
+		suggestion = suggestion + ' these group of columns have different results than initial columns so you might also look for the given group of columns'
+		return suggestion
+	else:
+		return ""
+
+def check_dominent_percentage(table, dimensions, slice_compare_column, 
```

Comment:
Prefix this function with a _ since its an internal function: http://google.github.io/styleguide/pyguide.html#3164-guidelines-derived-from-guidos-recommendations